streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatter.java

```diff
@@ -150,7 +152,15 @@ interface SegmentValue {
          */
         SegmentSearchResult find(long timestamp, boolean includeValue);
 
-        List<SegmentSearchResult> findAll(long fromTime, long toTime);
+        /**
+         * Finds the next/previous record in this segment row that is valid within the query specified time range
+         *
+         * @param fromTime the query starting time point
+         * @param toTime the query ending time point
+         * @param order specifies the order (based on timestamp) in which records are returned
+         * @return the record that is found, null if no record is found
+         */
+        SegmentSearchResult find(long fromTime, long toTime, ResultOrder order);
```

Comment:
I am wondering if existing `find()` and the new `find()` can be called interleaved without breaking anything?

So far, `PartiallyDeserializedSetmentValue` is already lazy and deserializes from newest to oldest. It would be a difficult API contract if calling existing `find` and new `find` interleaved would break anything?

Reply:
> I am wondering if existing `find()` and the new `find()` can be called interleaved without breaking anything?

When does that happen? Right now, only instances of `LogicalSegmentIterator` have access to the new `find` method; therefore, I think the same instance of `PartiallyDeserializedSetmentValue` cannot call them interleaved.